### PR TITLE
ci: migrate release to stable branch

### PR DESCRIPTION
**Pull request by amannocci**
## What does this PR do?

* Migration of the release stable branch to GitHub Actions.
* The sync between main and stable branch will be handled by the GitHub Actions.

## Validation

Ref: https://github.com/elastic/apm-agent-java/actions/runs/4502205643/jobs/7923696347

### CODE REVIEW

File: .github/workflows/sync-branches.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  sync:
+    name: Sync branches
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Synchronize source to target branch
+        run: |
+          git checkout -f "${SOURCE_BRANCH}"
```
Comment (v1v) [inline .github/workflows/sync-branches.yml:18]: I somehow feel nervous to use env variables, even though they were already used in Jenkins.

**Review by v1v — COMMENTED** [review]
I have not tested it out, but wonder if this works with the current GitHub token permissions, IIRC, it's read only based.

**Comment by amannocci** [comment]
> I have not tested it out, but wonder if this works with the current GitHub token permissions, IIRC, it's read only based.

By default, a workflow has read/write permissions scope in permissive mode.
It should be working, though it could be worth using the restricted mode as default and granting permissions on demand.

File: .github/workflows/sync-branches.yml (hunk 1 above)
Comment (v1v) [reply .github/workflows/sync-branches.yml:18]: That's fine.

Can you confirm whether this step worked in GH action with the [current default read permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)?

Comment (amannocci) [reply .github/workflows/sync-branches.yml:18]: AFAIS, we use the default permissive mode, which grants read/write permissions for all scopes.
I tested the workflow using a different source and target branch.
https://github.com/elastic/apm-agent-java/actions/runs/4502205643/jobs/7923696347

Comment (v1v) [reply .github/workflows/sync-branches.yml:18]: Cool, I'd say to put the explicit permissions just in case those are changed in the future.

**Review by v1v — APPROVED** [review]
Thanks

**Pull request merged**